### ISSUE TITLE
npm run build broken on npm 6.10 on MacOSX 

### ISSUE THREAD

**Issue by niebert** [spencermountain/wtf_wikipedia#101]
`npm run build` is broken on MacOSX with npm 6.10. Building the `wtf_wikipedia.js` and  `wtf_wikipedia.min.js` does not contain the javascript code, just the headers and tails of the module. Probably `browserify` seems not to be called OS independently.

**Comment by spencermountain** [comment]
yeah, sorry. this is a browserify thing. it reads the "browser" field from the package.json and then decides to suck.
it was a drag getting the browser/server fetch/promise thing working, and that was as close as i could get.

**Comment by niebert** [comment]
No worries, you do a great job no need to say sorry. The major revision provides in the end a major improvement. You could do me great favor, if you could consider to make the AST abstract syntax tree of the wtf parsing available in a wtf attribute e.g. `wtf.ast` which could be JSON similar to the following example. Exporting the AST is determined by the tree node type `sentence`, `table` and the output handlers that you have already in `/src/output/` for the formats. The available JSON Abstract Syntax Tree will make the whole export format approach (like PanDoc https://www.pandoc.org/try ) even more modular. Developers use the AST to generate output formats beyond the MarkUp, Plain, LaTeX and HTML. without the need to touch `wtf_wikipedia`. Keeping the currently available basic export formats in wtf_wikipedia makes a lot of sense, but beyond developers should use the AST to generate more output formats from the Wiki sources.  Will add the following example to the GitHub-Wiki to discuss the format further prior to any further implementation.
```javascript
{
   language: "en",
   domain: "wikiversity",
   article: "Water",
   ast: [
       {
           type:"paragraph",
           value:"",
           children:[
              {
                   type:"sentence",
                   value:"My first sentence.",
                  children:[]
             },
           {
                   type:"sentence",
                   value:"My Second sentence.",
                  children:[]
             },
          ]
       },
       {
           type:"title",
           value:"My Title",
           children:null

       },
       {
           type:"math",
           value:"\sum_{k=1}^{n} k^2",
           children:[]

       },
    ]
}
```

**Comment by spencermountain** [comment]
hey Engelbert, is this the formal AST syntax?  ... is there a formal AST syntax?
i'd be happy to include a formal AST in the outputs, you can see it now has a [json output](https://github.com/spencermountain/wtf_wikipedia/tree/master/src/output/json), so it wouldn't be hard to shape that in another way.

i don't know how we'd handle all the links and stuff, it sounds a little brainy, and since it's handled by pandoc already, maybe outside the scope of this library? I dunno, i'm not against it, i could be convinced.
cheers

**Comment by spencermountain** [comment]
if you have a really ambitious view for representing wikipedia articles as an AST, you could write a library that takes a `Document` object, and outputs an AST pretty easily - that sort of thing should become really easy to do, with this new API.

**Comment by niebert** [comment]
Thank you very much, you have done the major work to enable the AST handling that with the output handler for sentence, sections,....  Appreciate you refactoring of wtf_wikipedia very much.
 AST: I am exploring different existing approaches, Uglify seems to have a good generic structure http://lisperator.net/uglifyjs/parser . PanDoc does a good job, but is does not run in the browser. 

I regard the AST as an evolutionary (suggestion for wtf_wikipedia) approach to decompose the wikistructure as good as possible. Generated a AST sample, that seems to me close to the src/output handlers. 

Thank you for the new API. Will it be OK you to write general concepts of application in the Wiki and then discuss, what developement part should be within wtf_wikipedia or build a new library around wtf_wikipedia. Chaining of data extractions into JSON / document generation 

Thank you very much, Cheers Bert

**Comment by spencermountain** [comment]
hey, cool. Yeah sounds great.
I'm committed to getting it really strong over the next few months. Having people extend it in clever ways should be a central focus.

**Comment by niebert** [comment]
Full support for that!

**Comment by niebert** [comment]
Perfect, thank you very much
